Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticTypes.cs

```diff
@@ -42,7 +39,77 @@ public struct ElasticExtentProperties<T>
         /// Points inside the extent to which the system will snap.
         /// </value>
         [SerializeField]
-        public T[] SnapPoints;
+        public float[] SnapPoints;
+
+        /// <value>
+        /// Distance at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float SnapRadius;
+    }
+
+    /// <summary>
+    /// Properties of a three-dimensional extent
+    /// in which a damped harmonic oscillator is free to move.
+    /// </summary>
+    [Serializable]
+    public struct VolumeElasticExtent
+    {
+        /// <value>
+        /// Represents the lower bound of the extent.
+        /// </value>
+        [SerializeField]
+        public Bounds StretchBounds;
+
+        /// <summary>
+        /// Whether the bounds should be respected by the system.
+        /// </summary>
+        [SerializeField]
+        public bool UseBounds;
+
+        /// <value>
+        /// Points inside the extent to which the system will snap.
+        /// </value>
+        [SerializeField]
+        public Vector3[] SnapPoints;
+
+        /// <value>
+        /// Should the SnapPoints be repeated to infinity to serve
+        /// as regular snapping intervals?
+        /// </value>
+        public bool repeatSnapPoints;
+
+        /// <value>
+        /// Distance at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float SnapRadius;
+    }
+
+    /// <summary>
+    /// Properties of a four-dimensional extent
+    /// in which a damped harmonic oscillator is free to rotate.
+    /// </summary>
+    [Serializable]
+    public struct QuaternionElasticExtent
+    {
+        /// <value>
+        /// Euler angles to which the system will snap.
+        /// </value>
+        [SerializeField]
+        public Vector3[] SnapPoints;
+
+        /// <value>
+        /// Should the SnapPoints be repeated to infinity to serve
+        /// as regular snapping intervals?
+        /// </value>
+        public bool repeatSnapPoints;
```

Comment:
What does repeating to infinity mean within the context of a set of euler angles? (Or just, repeated until it fits the entire sphere?)